packages/@aws-cdk/cloudformation-include/lib/short-form-tags.ts

```diff
@@ -0,0 +1,160 @@
+          yaml.parse((cstNode.toString().substring(lastDot + 1))),
+        ],
+      };
+    }
+
+    return {
+      'Fn::GetAtt': (yaml.parse(cstNode.toString().substring('!GetAtt'.length))),
+    };
+  },
+};
+const getAZs = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!GetAZs',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::GetAZs': (yaml.parse(cstNode.toString().substring('!GetAZs'.length))),
+    };
+  },
+};
+const importValue = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!ImportValue',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::ImportValue': (yaml.parse(cstNode.toString().substring('!ImportValue'.length))),
+    };
+  },
+};
+const join = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!Join',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::Join': (yaml.parse(cstNode.toString().substring('!Join'.length))),
+    };
+  },
+};
+const select = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!Select',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::Select': (yaml.parse(cstNode.toString().substring('!Select'.length))),
+    };
+  },
+};
+const split = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!Split',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::Split': (yaml.parse(cstNode.toString().substring('!Split'.length))),
+    };
+  },
+};
+const transform = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!Transform',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::Transform': (yaml.parse(cstNode.toString().substring('!Transform'.length))),
+    };
+  },
+};
+const fnAnd = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!And',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::And': (yaml.parse(cstNode.toString().substring('!And'.length))),
+    };
+  },
+};
+const fnEquals = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!Equals',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::Equals': (yaml.parse(cstNode.toString().substring('!Equals'.length))),
+    };
+  },
+};
+const fnIf = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!If',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::If': (yaml.parse(cstNode.toString().substring('!If'.length))),
+    };
+  },
+};
+const fnNot = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!Not',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::Not': (yaml.parse(cstNode.toString().substring('!Not'.length))),
+    };
+  },
+};
+const fnOr = {
+  identify: (value: any) => typeof value === 'string',
+  tag: '!Or',
+  resolve: (_doc: any, cstNode: any) => {
+    return {
+      'Fn::Or': (yaml.parse(cstNode.toString().substring('!Or'.length))),
+    };
+  },
+};
```

Comment:
This is an insaaaaaaane amount of duplication that's going on here 🙂. Here's one idea on how to fix it:

```ts
function makeTagForCfnIntrinsic(intrinsicName: string, addFnPrefix: boolean = true,
  resolveFun?: (_doc: yaml.Document, cstNode: yaml_cst.CST.Node) => any): yaml_types.Schema.Tag {

  return {
    identify(value: any) { return typeof value === 'string'; },
    tag: `!${intrinsicName}`,
    resolve: resolveFun || ((_doc: yaml.Document, cstNode: yaml_cst.CST.Node) => {
      const ret: any = {};
      ret[addFnPrefix ? `Fn::${intrinsicName}` : intrinsicName] =
        // the +1 is to account for the ! the short form begins with
        yaml.parse(cstNode.toString().substring(intrinsicName.length + 1));
      return ret;
    }),
  };
}

export const shortForms: yaml_types.Schema.Tag[] = [
  'Base64', 'Cidr', 'FindInMap', 'GetAZs', 'ImportValue', 'Join',
  'Select', 'Split', 'Transform', 'And', 'Equals', 'If', 'Not', 'Or',
].map(name => makeTagForCfnIntrinsic(name)).concat(
  makeTagForCfnIntrinsic('Ref', false),
  makeTagForCfnIntrinsic('GetAtt', true, (_doc: yaml.Document, cstNode: yaml_cst.CST.Node): any => {
    const lastDot = cstNode.toString().lastIndexOf('.');

    return {
      'Fn::GetAtt': lastDot !== -1
        ? [
          cstNode.toString().substring('!GetAtt '.length, lastDot),
          yaml.parse((cstNode.toString().substring(lastDot + 1))),
        ]
        : yaml.parse(cstNode.toString().substring('!GetAtt'.length)),
    };
  }),
);
```

Reply:
Thanks, that's a much better solution!